### ISSUE TITLE
Unexpected --user on kool exec breaks with permission error

### ISSUE THREAD

**Issue by fabriciojs** [kool-dev/kool#166]
`kool exec` in a project with official `php:7.4.2-fpm-alpine` breaks. That is because we enter the image with a non-existing user, then any write will fail with permission issues:

```
➜  kool git:(master) docker run --rm -d php:7.4.2-fpm-alpine                                                               
64c6f469e24728d1ddbef962a6cb3c6b1bd06bc49bb552fd32376db06a2d992d
➜  kool git:(master) # works     
➜  kool git:(master) docker exec 64c6f469e24728d1ddbef962a6cb3c6b1bd06bc49bb552fd32376db06a2d992d touch /tmp/xxx
➜  kool git:(master) # do not work
➜  kool git:(master) docker exec --user 501 64c6f469e24728d1ddbef962a6cb3c6b1bd06bc49bb552fd32376db06a2d992d touch /tmp/xxx
touch: /tmp/xxx: Permission denied
```

Something created with root or any other user, when trying to access it with a non-existing user, will fail.

Currently we will always define `KOOL_ASUSER` (with current user ID), and then inject it:

https://github.com/kool-dev/kool/blob/master/cmd/exec.go#L55

We need to figure out the better UX to accommodate images that support this and images that do not.

**Comment by fabriciojs** [comment]
One idea is:

- We never add the `--user` on kool exec. We add some support on the kool.yml presets file to come from there the definition. This is still not 100% because people may use the default presets with custom images, which would still cause issues.

**Comment by danielsuguimoto** [comment]
@fabriciojs can't we do something that might make our own images unique somehow? Then we could detect it as we do on `kool docker` command?

**Comment by fabriciojs** [comment]
@danielsuguimoto that is one way we can explore, but has its flaws... 

`kool docker` like `kool exec` have issues with options ordering when proxying to docker/docker-compose... For instance I currently have use cases where I need more flags to be send to `docker run` so I do like:

`kool docker -- --network xxx image:tag command`

So I am able to define the `network` even though we did not create such option. But in this case we no longer no what of the arguments in the list is the action target image.

Note: `--` as a parameters tells the command to stop parsing any flags, start treating them all as arguments.

We need another way to "check" if we need/want or not to set the user. UX-wise. That  is the big question indeed.

**Comment by fbrissi** [comment]
It's the same problem I had in a other project with Symfony, in my case it was because of Windows.

On Windows it is always set to root (0), so the files created by the composer for example, fpm did not have permission.

I do an fix that I'm still testing and seeing if it resolved 100%, instead of windows setting root (0), Setei a fixed user 9999.
This solved this problem, as the composer started to write with a user in the same group as the kool user.

In the case of windows I had thought of leaving it fixed in some value other than 0 to not be root, or take the last numbers of the SID (S-1-5-21-3623811015-3361044348-30300820-**1013**).
But I don't now if I could have more problems with that, in other images.

But I think that this type of treatment can be difficult to control.
Because we don't just have the images of the kool.

If the kool image runs with root, it would not prevent this kind of problem?

**Comment by danielsuguimoto** [comment]
https://jtreminio.com/blog/running-docker-containers-as-current-host-user/

**Comment by fabriciojs** [comment]
@fbrissi 

> If the kool image runs with root, it would not prevent this kind of problem?

So we actually have the different user to solve the permissions problem for PHP/JS Kool provided images.

I am thinking that maybe this should be a feature toggle, within `kool.yml`, a new root level key `user: [ root | $UID ]`, then every time we can pass in --user to docker or docker-compose, we do pass in such value.

**Comment by fabriciojs** [comment]
Anyone reading through this please see the attached discussion that took place at https://github.com/kool-dev/kool/issues/216 as well.

**Comment by fabriciojs** [comment]
A simple adaptation for custom images is adding to Dockerfile:

```
adduser -g 0 kool
```
 
This has been used in a couple cases on `archlinux` based images, so it may vary due to distro specifics, but all in all, adding the user is the best workaround at this moment.

The other well-used workaround when we know we want to use default `root` for commands execution is to add to your `.env` file:

```
KOOL_ASUSER=0
```

**Comment by fabriciojs** [comment]
Implemented new pre-check on #243 - which covers this for now!
